CMakeLists.txt

```
@@ -1705,7 +1705,9 @@ if(CONFIG_OUTPUT_DISASSEMBLY)
 if(CONFIG_OUTPUT_DISASSEMBLE_ALL)
     set(disassembly_type "$<TARGET_PROPERTY:bintools,disassembly_flag_all>")
   else()
-    set(disassembly_type "$<TARGET_PROPERTY:bintools,disassembly_flag_inline_source>")
+    if (CONFIG_OUTPUT_DISASSEMBLE_WITH_SOURCE)
+        set(disassembly_type "$<TARGET_PROPERTY:bintools,disassembly_flag_inline_source>")
+    endif()
   endif()
```

why the extra `if()` ?
```suggestion
  elseif (CONFIG_OUTPUT_DISASSEMBLE_WITH_SOURCE)
    set(disassembly_type "$<TARGET_PROPERTY:bintools,disassembly_flag_inline_source>")
  endif()
```